experimental/sql/dialects/rel_impl.py

```diff
@@ -482,6 +482,39 @@ def get(input: Operation, predicates: Region) -> 'Select':
                          result_types=[input.results[0].typ])
 
 
+@irdl_op_definition
+class NestedLoopCartesianProduct(Operator):
```

Comment:
I am not happy with this name. i wanted to express, that this is a lower lveel than rel_ssa, but I feel that nested loops is probably the most sensible way to implement Cartesian products anyhow, so defining it here feels out of place. I think this could just (again) be call `cartesian_product`. What do you think, @ingomueller-net?